src/api/likes.ts

```diff
@@ -0,0 +1,62 @@
+import { db } from '@/api/firebaseApp'
+import { doc, getDoc, updateDoc, setDoc, deleteDoc } from 'firebase/firestore'
+
+export const fetchLikes = async (playlistId: string) => {
+  const docRef = doc(db, 'Likes', playlistId)
+  const docSnap = await getDoc(docRef)
+
+  if (docSnap.exists()) {
+    const data = docSnap.data()
+    return {
+      likeCount: Number(data.likeCount) || 0,
+      userLikes: data.userLikes || {}
+    }
+  } else {
+    return { likeCount: 0, userLikes: {} }
+  }
+}
+
+export const updateLikes = async (
+  playlistId: string,
+  userId: string,
+  isLiked: boolean
+) => {
+  const docRef = doc(db, 'Likes', playlistId)
+
+  const docSnap = await getDoc(docRef)
+  let newLikeCount
+
+  if (docSnap.exists()) {
+    const currentData = docSnap.data()
+
+    if (isLiked) {
+      newLikeCount = (currentData.likeCount || 0) + 1
+      await updateDoc(docRef, {
+        [`userLikes.${userId}`]: true,
+        likeCount: newLikeCount
+      })
+    } else {
+      newLikeCount = Math.max(0, (currentData.likeCount || 0) - 1)
```

Comment:
P3;

코드는 직관적인 것이 좋다고 생각합니다. 아래처럼 삼항연산자로 로직을 직관적으로 표현하면 어떨까요?
원래코드는 currentData.likeCount에서 무조건 1을 뺀 뒤, max 메서드로 0과 비교해 더 큰 값을 내는데,
"like 카운트가 1 이상일 때는 -1을 하고, 0이면 하지 않는다"는 의미를 직관적으로 표현하진 못하는 것 같습니다.
```suggestion
      newLikeCount = currentData.likeCount > 0 ? currentData.likeCount - 1 : 0;
```

Reply:
물론 좀 더 가독성을 생각한다면 아래와 같은 방법도 좋습니다.
```suggestion
      newLikeCount = currentData.likeCount
      if (newLikeCount > 0) {
        newLikeCount -= 1;
      }
```

Reply:
확실히 더 보기 편해진 것 같습니다! 감사합니다!

Reply:
멘토님 말씀대로 `newLikeCount = currentData.likeCount > 0 ? currentData.likeCount - 1 : 0`가 훨씬 직관적인 것 같네요. 감사합니다!